relatedness/get_ibd.py

```diff
@@ -0,0 +1,407 @@
+    sex_check = args.sex_check
+    mt_path = args.mt_path
+    input_pedigree = args.input_pedigree
+
+    gnomad_ld = args.gnomad_ld
+    run_ibd = args.run_ibd
+    first_degree_pi_hat = args.first_degree_pi_hat
+    grandparent_pi_hat = args.grandparent_pi_hat
+    grandparent_ibd1 = args.grandparent_ibd1
+    grandparent_ibd2 = args.grandparent_ibd2
+
+    logger.info("Reading in inputs...")
+    mt = hl.read_matrix_table(mt_path)
+    pedigree = hl.import_table(input_pedigree, impute=True)
+
+    # Infer build of the MatrixTable
+    build = get_reference_genome(mt.locus).name
+
+    logger.info("Filtering to biallelic SNVs on autosomes and performing LD pruning...")
+    mt = filter_rows_for_qc(mt, min_af=0.001, min_callrate=0.99, apply_hard_filters=False)
+    mt = ld_prune(mt, build, gnomad_ld)
+    out_mt = f'{output_dir}/{output_name}_processed_mt.mt'
+
+    logger.info(
+        "Reading sex check into dictionary..."
+    )  
+    # Code here is based on old sex check, need new sex check that works on dense data
+    sample_sexes = {}
+    with hl.hadoop_open(sex_check, "r") as infile:
+        next(infile)
+        for line in infile:
+            line = line.rstrip()
+            items = line.split("\t")
+            sample = items[0]
+            sex = items[6]
+            if sex == "female":
+                sample_sexes[sample] = True
+            elif sex == "male":
+                sample_sexes[sample] = False
+
+    logger.info("Remapping sample names...")
+    mt, sample_sexes = remap_samples(
+        mt_path, mt, pedigree, sample_sexes
+    )
+
+    mt = mt.checkpoint(out_mt, overwrite=True)
+
+    if run_ibd:
+        logger.info("Running identity by descent...")
+        ibd_results_ht = hl.identity_by_descent(mt, maf=mt.AF, min=0.10, max=1.0)
+        ibd_results_ht = ibd_results_ht.annotate(
+            ibd0=ibd_results_ht.ibd.Z0,
+            ibd1=ibd_results_ht.ibd.Z1,
+            ibd2=ibd_results_ht.ibd.Z2,
+            pi_hat=ibd_results_ht.ibd.PI_HAT,
+        ).drop("ibs0", "ibs1", "ibs2", "ibd")
+        out_ht = f"{output_dir}/{output_name}_ibd_kinship.tsv"
+        ibd_results_ht.export(out_ht)
+
+    else:
+        logger.warn("Skipping IBD - using previous calculations...")
+        if not file_exists(f'{output_dir}/{output_name}_ibd_kinship.tsv'):
+            logger.warning(
+                "IBD calculation was skipped but no file with previous calculations was found...",
+                sample,
+                )
+
+    logger.info("Reading in kinship ht...")
+    kin_ht = hl.import_table(
+        f'{output_dir}/{output_name}_ibd_kinship.tsv', impute=True
+    )
+
+    # Subset MatrixTable to the samples in the pedigree
+    mt_subset, expected_samples, vcf_samples = subset_mt(
+        mt, pedigree, output_dir, output_name
+    )
+    num_ped = len(expected_samples)
+    num_vcf = len(vcf_samples)
+
+    # Subset Table to the samples in the pedigree
+    subset = hl.literal(expected_samples)
```

Comment:
I think we'll also want a set here? I think it's currently an array
```suggestion
    subset = hl.set(expected_samples)
```